crates/sui-types/src/crypto.rs

```diff
@@ -325,6 +327,11 @@ impl EncodeDecodeBase64 for PublicKey {
                         FastCryptoError::InputLengthWrong(Secp256r1PublicKey::LENGTH + 1),
                     )?)?;
                     Ok(PublicKey::Secp256r1((&pk).into()))
+                } else if x == &SignatureScheme::PasskeyAuthenticator.flag() {
+                    let pk = Secp256r1PublicKey::from_bytes(bytes.get(1..).ok_or(
+                        FastCryptoError::InputLengthWrong(Secp256r1PublicKey::LENGTH + 1),
+                    )?)?;
+                    Ok(PublicKey::Passkey((&pk).into()))
```

Comment:
passkey supports one other signature scheme eddsa, but since recoverability (derive pk from signature) is a hard requirement, the only scheme we can use from passkey is r1 so we are hardcoding it here